GitCommands/Git/GitGpgController.cs

```diff
@@ -24,12 +25,30 @@ public enum TagStatus
 
     public interface IGitGpgController
     {
+        /// <summary>
+        /// Gets the user's default signing key from git config
+        /// </summary>
+        /// <returns>The key id of the gpg key to sign with.</returns>
+        Task<string> GetDefaultKeyAsync();
+
         /// <summary>
         /// Obtain the commit signature status on current revision.
         /// </summary>
         /// <returns>Enum value that indicate the gpg status for current git revision.</returns>
         Task<CommitStatus> GetRevisionCommitSignatureStatusAsync(GitRevision revision);
 
+        /// <summary>
+        /// Gets whether tags are signed by default
+        /// </summary>
+        /// <returns> true for default to signed tags</returns>
+        Task<bool> GetTagGPGSignAsync();
+
+        /// <summary>
+        /// Gets whether commits are signed by default
+        /// </summary>
+        /// <returns> true for default to signed commits</returns>
+        Task<bool> GetCommitGPGSignAsync();
```

Comment:
I'm not really excited by having multiple new public API.... Have you considered adding only a single method that would read the whole lot of GPG-related settings and return those as a bundle (e.g., a struct)?

Reply:
The reason is that I avoid calling multiple configs when I only need the piece that I need.  Tjis controller already existed and it drives the  things you do with GPG.  If i do as you suggest then I will have to call all git config options even though I only need x item at the time.

- commit.gpgSign
- tag.gpgSign
- gpg.program
- user.signingKey

No I cannot just use the config settings cache because I must get current config at the time of run.  So this allows me to ask for what I want when I need it.  If I need tag.gpgSign then i ask for it.  If I need the default key... and so on.  Just because this controller only had gpg signature display items, it made sense for this to drive the whole functionality of GPG info.